include/rapidjson/document.h

```diff
@@ -1235,8 +1235,8 @@ class GenericValue {
             // return NullValue;
 
             // Use static buffer and placement-new to prevent destruction
-            static char buffer[sizeof(GenericValue)];
-            return *new (buffer) GenericValue();
+            static GenericValue buffer;
+            return *new (reinterpret_cast<char *>(&buffer)) GenericValue();
```

Comment:
I don't think using placement-new here is thread-safe, and it will (rarely) cause undefined behavior. The default constructor of `GenericValue` can be executed by different threads simultaneously and each of them initializes its members on the same memory because `buffer` is declared as `static`.